posthog/hogql_queries/insights/trends/trends_query_runner.py

```diff
@@ -309,3 +197,22 @@ def apply_formula(self, formula: str, results: List[Dict[str, Any]]) -> List[Dic
         new_result["label"] = f"Formula ({formula})"
 
         return [new_result]
+
+    def _is_breakdown_field_boolean(self):
+        field_type = self._event_properties.get(self.query.breakdown.breakdown)
+        return field_type == "Boolean"
+
+    def _convert_boolean(self, value: any):
+        bool_map = {1: "true", 0: "false", "": ""}
+        return bool_map.get(value) or value
+
+    @cached_property
+    def _event_properties(self):
+        event_property_values = PropertyDefinition.objects.filter(
+            team_id=self.team.pk,
+            type__in=[None, PropertyDefinition.Type.EVENT],
+        ).values_list("name", "property_type")
+
+        event_properties = {name: property_type for name, property_type in event_property_values if property_type}
+
+        return event_properties
```

Comment:
Some teams, like ours, have millions of event properties due to old plugins. Maybe we don't want all of them in memory 😬 

<img width="567" alt="image" src="https://github.com/PostHog/posthog/assets/53387/86782005-90a4-429c-8024-9c46f215c1fd">

Reply:
Yep, that's fair - have changed this to a `get` on the individual field as opposed to `get all and then filter`